ui/app/src/components/BrowseFilesDialog/BrowseFilesDialogUI.tsx

```diff
@@ -63,33 +80,151 @@ export function BrowseFilesDialogUI(props: BrowseFilesDialogUIProps) {
             <FolderBrowserTreeView rootPath={path} onPathSelected={onPathSelected} selectedPath={currentPath} />
-          </section>
+          </Box>
           <section className={classes.rightWrapper}>
-            <InputUnstyled value={currentPath} className={classes.currentPath} disabled title={currentPath} />
-            <Divider />
-            <Box display="flex" alignItems="center" marginTop="16px" marginBottom="16px" gap="8px">
-              {allowUpload && (
-                <SecondaryButton onClick={onUpload}>
-                  <FormattedMessage id="word.upload" defaultMessage="Upload" />
-                </SecondaryButton>
-              )}
-              <SecondaryButton onClick={onRefresh}>
-                <FormattedMessage id="word.refresh" defaultMessage="Refresh" />
-              </SecondaryButton>
-              <SearchBar
-                keyword={keyword}
-                onChange={handleSearchKeyword}
-                showDecoratorIcon
-                showActionButton={Boolean(keyword)}
-                classes={{ root: classes.searchRoot }}
-              />
-            </Box>
+            <Paper className={classes.actionsBar}>
+              <Toolbar disableGutters>
+                <Box sx={{ flexGrow: 1, display: 'flex' }}>
+                  <Tooltip title={<FormattedMessage id="word.refresh" defaultMessage="Refresh" />}>
+                    <IconButton onClick={onRefresh}>
+                      <RefreshIcon />
+                    </IconButton>
+                  </Tooltip>
+                  {allowUpload && (
+                    <Tooltip title={<FormattedMessage id="word.upload" defaultMessage="Upload" />}>
+                      <IconButton onClick={onUpload} sx={{ mr: 1 }}>
+                        <UploadFileIcon />
+                      </IconButton>
+                    </Tooltip>
+                  )}
+                  <Divider orientation="vertical" flexItem className={classes.actionsBarDivider} />
+                  <SearchBar
+                    keyword={keyword}
+                    onChange={handleSearchKeyword}
+                    showDecoratorIcon
+                    showActionButton={Boolean(keyword)}
+                    classes={{ root: classes.searchRoot }}
+                  />
+                  <Divider orientation="vertical" flexItem className={classes.actionsBarDivider} />
+                  <Button
+                    id="sort-button"
+                    aria-haspopup="true"
+                    aria-expanded={sortMenuOpen ? 'true' : undefined}
+                    onClick={() => setSortMenuOpen(!sortMenuOpen)}
+                    ref={buttonRef}
+                    sx={{ ml: 1, mr: 2 }}
+                  >
+                    <FilterListIcon sx={{ mr: 1 }} />
+                    <FormattedMessage id="words.sorting" defaultMessage="Sorting" />
+                  </Button>
+                  <Menu
+                    id="basic-menu"
```

Comment:
make your own id.
i think the button is missing the aria-controls reference to this menu? look up the right aria thingie